test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/jbpm/ProcessFailoverIntegrationTest.java

```diff
@@ -111,6 +108,79 @@ public void processFailoverTest() {
         assertThat(deploymentScenario.getKieServerDeployment().getInstances().iterator().next()).isNotEqualTo(kieServerInstance);
     }
 
+    @Test
+    public void processFailoverRedirectionTest() {
+        registerKieContainer();
+
+        logger.debug("Get Kie Server Instance");
+        Instance kieServerInstance = deploymentScenario.getKieServerDeployment().getInstances().iterator().next();
+
+        Long longScriptPid = startProcessInstance(LONG_ASYNC_SCRIPT_PROCESS_ID);
+
+        assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
+        assertProcessVariable(longScriptPid, variableKey, variableValueOne);
+
+        logger.debug("Send signal to continue with process.");
+        signalStartLongScript(longScriptPid);
+
+        logger.debug("Scale Kie Server to 2");
+        deploymentScenario.getKieServerDeployment().scale(2);
+        logger.debug("Wait for scale");
+        deploymentScenario.getKieServerDeployment().waitForScale();
+
+        logger.debug("Force delete (Kill) Kie server instance.");
+        deploymentScenario.getKieServerDeployment().deleteInstances(kieServerInstance);
+        logger.debug("Wait for scale");
+        deploymentScenario.getKieServerDeployment().waitForScale();
+
+        waitUntilProcessVariableIsChanged(longScriptPid);
+
+        assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
+        assertProcessVariable(longScriptPid, variableKey, variableValueTwo);
+        processServicesClient.signalProcessInstance(CONTAINER_ID, longScriptPid, SIGNAL_2_NAME, null);
+        assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED);
+
+        assertThat(deploymentScenario.getKieServerDeployment().getInstances()).doesNotContain(kieServerInstance);
+    }
+
+    private void waitUntilProcessVariableIsChanged(Long pid) {
+        LocalDateTime endTime = LocalDateTime.now().plusMinutes(3);
+
+        while (LocalDateTime.now().isBefore(endTime)) {
+            ProcessInstance pi = null;
+            try {
+                pi = processServicesClient.getProcessInstance(CONTAINER_ID, pid, true);
+            } catch (KieServicesException ex) {
+                //ok
+            }
+
+            if (pi != null && pi.getVariables().containsValue(variableValueTwo)) {
```

Comment:
This condition should be moved into the try block. In case getProcessInstance() throws an exception then this code is irrelevant.